src/plugins/_manager.ts

```diff
@@ -0,0 +1,171 @@
+import * as child from 'child_process';
+import * as os from 'os';
+import * as path from 'path';
+import * as url from 'url';
+import * as fs from 'fs-extra';
+
+const MODULE_NOT_FOUND_ERROR_CODE = 'MODULE_NOT_FOUND'; // TODO is there a known constant we can use here?
+const MODULE_VERSION_MISMATCH_ERROR_CODE = 'MODULE_VERSION_MISMATCH';
+
+/**
+ * Error indicating a module was found but with an unexpected version.
+ */
+export class ModuleVersionMismatchError extends Error {
+
+  public readonly code: string = MODULE_VERSION_MISMATCH_ERROR_CODE;
+
+  constructor(pkg: string, expected: string, actual: string) {
+    super(`Version mismatch for package ${pkg}. Found ${actual} but requested ${expected}`);
+  }
+}
+
+export interface Plugin {
+
+  readonly instance: any;
```

Comment:
There's a bunch of `any` in these interfaces here... and I'd really recommend getting out of the habit of using `any` and instead using `unknown`, which is much safer.

Reply:
We talked about using the `Validation` interface here to type the `instance` property. 
However, at this point in the code, we don't actually know it is indeed an instance of `Validation`, this is being validated later on. The plugin manager just returns an instance of something, its not aware of what specific plugin invoked it, and its up to the specific plugin to perform its validations and cast it to the concrete type:

https://github.com/cdk8s-team/cdk8s-cli/blob/547114a6e5809bfb8ab100cdd6f2b99126b56698/src/plugins/validation.ts#L365-L371

I'm going to switch this to `unknown`, just so that consumers are forced to check it, even if that checks amounts to casting to `any` themselves.